src/components/Grades.js

```diff
@@ -88,75 +78,78 @@ const getOverallCredits = credits => {
   return rows
 }
 
-const Grades = props => {
+const Grades = (props) => {
   const classes = useStyles()
-  const { t } = useTranslation()
   const { mobile } = props
-  const dispatch = useDispatch();
-  const credits = useSelector(state => state.credits.credits)
-  const credits_fetched = useSelector(state => state.credits.fetched)
-  const courses = useSelector(state => state.courses.courses)
-  const courses_fetched = useSelector(state => state.courses.fetched)
+  const credits = useSelector((state) => state.credits)
+  const courses = useSelector((state) => state.courses)
+  const courses_fetched = useSelector((state) => state.fetched)
+  const courses_error = useSelector((state) => state.error)
 
-  useEffect(() => {
-    dispatch(fetch_credits())
-  }, [dispatch])
+  if (courses_error || credits.length === 0) {
+    return (
+      <div className={classes.error}>
+        <ErrorMessages />
+      </div>
+    )
+  }
 
-  if (courses_fetched && credits_fetched) {
-      return (
-        <Card className={classes.card}>
-          <CardHeader
-            className={classes.classHeader}
-            title={
-              <Typography
-                tabIndex="0"
-                variant="h1"
-                className={classes.classHeaderSpan}
-                style={{ fontSize: '20px' }}
-              >
-                {t('gac')}
-              </Typography>
-            }
-          />
-          <CardContent className={mobile ? classes.content : null}>
-            <Table>
-              <TableHead>
-                <TableRow className={classes.tableHeader}>
-                  <TableCell className={classes.tableCell} scope="col">
-                    {t('level')}
-                  </TableCell>
-                  <TableCell className={classes.tableCell} scope="col">
-                    {t('grades')}
-                  </TableCell>
-                  <TableCell className={classes.tableCell} scope="col">
-                    GPA
-                  </TableCell>
-                </TableRow>
-              </TableHead>
-              <TableBody>{getOverallCredits(credits)}</TableBody>
-            </Table>
+  if (credits !== []) {
```

Comment:
Is this if check for type? If not is this needed even though you check the length of credits on [line 89](f563e617b0ac9a67390d725a3d3bb6c2R89)?

Reply:
This check isn't needed. Things are being done a bit different in #212 where I'm doing some more refactoring, so I think I'll just leave this as is for this PR if that's okay with you

Reply:
I think that'd be fine with KaJuan lol. I believe he messaged me after his reviewing saying things looked good to him. Addressing this in #212 seems reasonable